packages/verified-fetch/src/verified-fetch.ts

```diff
@@ -328,30 +332,59 @@ export class VerifiedFetch {
       }
     }
 
+    // we have to .stat before .cat so we can get the size and make sure byte offsets are calculated properly
+    if (byteRangeContext.isRangeRequest && byteRangeContext.isValidRangeRequest) {
+      stat = stat ?? await this.unixfs.stat(resolvedCID, {
```

Comment:
This stat operation is unnecessary - you can disambiguate the `terminalElement` type by examining the `.type` field:

```js
if (terminalElement.type === 'file') {
  // tsc now knows terminalElement is `UnixFSFile`
  byteRangeContext.fileSize = terminalElement.unixfs.fileSize()
}

```